easybuild/easyblocks/n/nvhpc.py

```diff
@@ -202,7 +212,17 @@ def sanity_check_step(self):
             'dirs': [os.path.join(prefix, 'compilers', 'bin'), os.path.join(prefix, 'compilers', 'lib'),
                      os.path.join(prefix, 'compilers', 'include'), os.path.join(prefix, 'compilers', 'man')]
         }
+
         custom_commands = ["%s -v" % compiler for compiler in compiler_names]
+
+        # compile minimal example using -std=c++20 to catch issue where it picks up the wrong GCC
+        # (as long as system gcc is < 9.0)
+        # see: https://github.com/easybuilders/easybuild-easyblocks/pull/3240
+        tmpdir = tempfile.mkdtemp()
+        write_file(os.path.join(tmpdir, 'minimal.cpp'), NVHPC_MINIMAL_EXAMPLE)
+        minimal_compiler_cmd = "cd %s && nvc++ -std=c++20 minimal.cpp -o minimal" % tmpdir
+        custom_commands.append(minimal_compiler_cmd)
```

Comment:
```suggestion
        if LooseVersion(self.version) >= LooseVersion('21'):
            # compile minimal example using -std=c++20 to catch issue where it picks up the wrong GCC
            # (as long as system gcc is < 9.0)
            # see: https://github.com/easybuilders/easybuild-easyblocks/pull/3240
            tmpdir = tempfile.mkdtemp()
            write_file(os.path.join(tmpdir, 'minimal.cpp'), NVHPC_MINIMAL_EXAMPLE)
            minimal_compiler_cmd = "cd %s && nvc++ -std=c++20 minimal.cpp -o minimal" % tmpdir
            custom_commands.append(minimal_compiler_cmd)
```
Maybe just exclude the test if it's older (probably doesn't have support for c++20 yet, based on failed reports)